### PR TITLE
Code refactor and small bugfixes

### PR DESCRIPTION
**Pull request by therealpeterpython**
I combined #3 and #4 to get the best of both worlds, and updated the code on top of that. The dependencies have changed, so I suspect the nix/poetry files need to be updated, but I have no idea how to do that. Maybe you could help @mmarx?

We could also add a license to the project to make the permitted type of use clearer, but I'm not sure which would be best.

### CODE REVIEW

File: archive_etherpads.py [hunk 1]
```diff
@@ -34,163 +45,173 @@ def confirm(prompt=None, resp=False):
...
+        #s = f"{url} -> {new_url}"
+        #logger.warning(f"got status code {r.status_code}, consider moving {url}")
+        #if False:  # confirm("give Time to save?", resp=True):
+        #    print(s)
+        #    confirm("Done?")
+        #    return [new_url]
+        #else:
+        #    return []
+        # TODO: Don't hard-code URLs
+        # TODO: What is the seenotrettung about?
```
Comment (nicoa) [inline archive_etherpads.py:128]: at the time we created this project, there were some issues with etherpad servers. If a pad broke, one could still obtain the content via some "hack" (history mode or similar) and copy-paste it into a new one (seenotrettung called that time, term was created/evolved during a plenum) which would be the go-to pad then.

**Comment by nicoa** [comment]
Looks really good (besides the mentioned points in your above comment), thanks a lot for the work here!

I think regarding dependencies that should be fine for now, as you did not add any new depedneices which aren't in the standard library.

Regarding license: good point, will think about adding one.

**Comment by mmarx** [comment]
> The dependencies have changed, so I suspect the nix/poetry files need to be updated, but I have no idea how to do that. Maybe you could help @mmarx?

`nix` takes the dependencies straight from poetry, so it suffices to update those (strictly speaking, this is not even required as you just removed some external dependencies, but this will speed up building and shrink the nix closure): simply remove `pandas` and `pathlib2` from [pyproject.toml](https://github.com/nicoa/koma_archive/blob/master/pyproject.toml#L17) and run `poetry update`, then make sure you commit the updated `poetry.lock`.

File: archive_etherpads.py (hunk 1 above)
Comment (therealpeterpython) [reply archive_etherpads.py:128]: That makes sense. Can I remove that part from the code then?

Comment (nicoa) [reply archive_etherpads.py:128]: depends if we're now on a stable etherpad server or anticipate that issue in the future again

**Review by nicoa — APPROVED** [review]

**Comment by nicoa** [comment]
Thanks a lot for your work! really nice improvements :)

**Pull request merged**